lib/reversetunnel/agent.go

```diff
@@ -592,6 +595,36 @@ func (a *agent) handleDrainChannels() error {
 	}
 }
 
+// GetClusterNetworkConfig gets the cluster networking config from the connected proxy.
+func (a *agent) GetClusterNetworkConfig(ctx context.Context) (types.ClusterNetworkingConfig, error) {
+	var cancel func()
+	ctx, cancel = context.WithTimeout(ctx, defaults.DefaultIOTimeout)
+	defer cancel()
+
+	channel, requests, err := a.client.OpenChannel(ctx, chanNetworkConfig, nil)
```

Comment:
Why isn't this a global request?

Reply:
Good idea, I got confused by the difference in behavior between global and channel requests and thought you couldn't send a reply payload to either. But turns out you can for global requests and just not channel requests. I will swap it out for a global request.

Reply:
Switched to global requests here [bb4b92b](https://github.com/gravitational/teleport/pull/24702/commits/bb4b92ba5056e3277f13b8a5f63f31d405b34e98)

Reply:
Whoops the reverse tunnel server needed its own global request handler [6563642](https://github.com/gravitational/teleport/pull/24702/commits/6563642a30d9528e66081973ac77ffc028654773)

Reply:
The fact that we've been silently swallowing global requests (completely screwing over any potential for sensible extensibility of the protocol) means that we can't use them reliably unless we know that the server at the other end is sane. 😢

It seems like the only "safe" thing we can do from the client side of the reverse tunnel is open a channel (which, when unknown, will get correctly rejected). What do you think about attempting to open a channel with a name like `reverse-tunnel-v2@goteleport.com` at the beginning of the connection and expecting it to be rejected with a custom rejection code to signify that the server will actually follow the SSH protocol (by rejecting unknown requests and handling requests that want a reply)?

Reply:
@espadolini Let me put something together 😄